packages/cloudflare/src/cli/build/index.ts

```diff
@@ -15,26 +21,84 @@ import { buildWorker } from "./build-worker";
  * @param projectOpts The options for the project
  */
 export async function build(projectOpts: ProjectOptions): Promise<void> {
-  if (!projectOpts.skipNextBuild) {
+  printHeader("Cloudflare build");
+
+  showWarningOnWindows();
+
+  const baseDir = projectOpts.sourceDir;
+  const require = createRequire(import.meta.url);
+  const openNextDistDir = dirname(require.resolve("@opennextjs/aws/index.js"));
+
+  const { config, buildDir } = await compileOpenNextConfig(baseDir);
+
+  ensureCloudflareConfig(config);
+
+  // Initialize options
+  const options = buildHelper.normalizeOptions(config, openNextDistDir, buildDir);
+  logger.setLevel(options.debug ? "debug" : "info");
+
+  // Pre-build validation
+  buildHelper.checkRunningInsideNextjsApp(options);
+  logger.info(`App directory: ${options.appPath}`);
+  buildHelper.printNextjsVersion(options);
+  buildHelper.printOpenNextVersion(options);
+
+  if (projectOpts.skipNextBuild) {
+    logger.warn("Skipping Next.js build");
+  } else {
     // Build the next app
-    await buildNextjsApp(projectOpts.sourceDir);
+    printHeader("Building Next.js app");
+    setStandaloneBuildMode(options);
+    buildNextjsApp(options);
   }
 
   if (!containsDotNextDir(projectOpts.sourceDir)) {
     throw new Error(`.next folder not found in ${projectOpts.sourceDir}`);
   }
 
-  // Clean the output directory
-  await cleanDirectory(projectOpts.outputDir);
+  // Generate deployable bundle
+  printHeader("Generating bundle");
+  buildHelper.initOutputDir(options);
+
+  // Compile middleware
+  await createMiddleware(options, { forceOnlyBuildOnce: true });
 
   // Copy the .next directory to the output directory so it can be mutated.
   cpSync(join(projectOpts.sourceDir, ".next"), join(projectOpts.outputDir, ".next"), { recursive: true });
 
-  const config = getConfig(projectOpts);
+  const projConfig = getConfig(projectOpts);
 
-  await buildWorker(config);
+  await buildWorker(projConfig);
+
+  logger.info("OpenNext build complete.");
 }
 
-async function cleanDirectory(path: string): Promise<void> {
-  return await rm(path, { recursive: true, force: true });
+/**
+ * Ensures open next is configured for cloudflare.
+ *
+ * @param config OpenNext configuration.
+ */
+function ensureCloudflareConfig(config: OpenNextConfig) {
+  const requirements = {
+    isExternal: config.middleware?.external == true,
+    useCloudflareWrapper: config.middleware?.override?.wrapper === "cloudflare",
+    useEdgeConverter: config.middleware?.override?.converter === "edge",
+    disableCacheInterception: config.dangerous?.enableCacheInterception !== true,
+  };
```

Comment:
Wouldn't it make sense for these to be default values if a user doesn't specify them in their config? As in, they're choosing the Cloudflare adapter, so their intention would be to want a project built for Cloudflare, unless they specifically put other config options.

Looking at the code in the aws adapter, it looks like the default behaviour is to only build an edge config if the middleware.external option is there - I think it would make sense for the cloudflare adapter's default to be to always build the edge config instead of requiring people add the extra config options, and then make sure that their config doesn't try to use an option that goes against the requirements for Cloudflare.

What do you think?

Reply:
Reading conico's comment again, this sounds like a similar point to what #114 is for, so marking as resolved.